File: vinzi.jsonMgt/src/main/java/vinzi/jsonMgt/hib/DbPatch.java
package vinzi.jsonMgt.hib;

public class DbPatch {
    public Long id;
    public Long getId() { return id;}
    protected void setId(Long id)  {this.id = id; };


    public Long track_id;
    public Long getTrack_id() { return track_id;}
    public void setTrack_id(Long track_id)  {this.track_id = track_id; };


    public java.util.Date datetime;
    public java.util.Date getDatetime() { return datetime;}
    public void setDatetime(java.util.Date datetime)  {this.datetime = datetime; };


    public String path;
    public String getPath() { return path;}
    public void setPath(String path)  {this.path = path; };


    public String action;
    public String getAction() { return action;}
    public void setAction(String action)  {this.action = action; };


    public String patchkey;
    public String getPatchkey() { return patchkey;}
    public void setPatchkey(String patchkey)  {this.patchkey = patchkey; };


    public String value;
    public String getValue() { return value;}
    public void setValue(String value)  {this.value = value; };


    public DbPatch() {};

    public DbPatch(Long id, Long track_id, java.util.Date datetime, String path, String action, String patchkey, String value) {
	this.id = id;
	this.track_id = track_id;
	this.datetime = datetime;
	this.path = path;
	this.action = action;
	this.patchkey = patchkey;
	this.value = value;
	return;};

   public String toString() {
        return " id=" + this.id + " track_id=" + this.track_id + " datetime=" + this.datetime + " path=" + this.path + " action=" + this.action + " patchkey=" + this.patchkey + " value=" + this.value;};

};

File: vinzi.jsonMgt/src/main/clojure/vinzi/jsonMgt/hib/clj/DbPatch.clj
(ns vinzi.jsonMgt.hib.clj.DbPatch
    (:use vinzi.hib-connect.globals)
    (:import vinzi.jsonMgt.hib.DbPatch))

(defrecord DbPatch_clj [id track_id datetime path action patchkey value])

(defn create-DbPatch [track_id datetime path action patchkey value]
  ;; TO DO:  insert type-checking (java-objects are strictly typed)
  (DbPatch_clj. nil track_id datetime path action patchkey value))

(defn clone-DbPatch [cRec]
  ;; TO DO:  insert type-checking cRec should be same type/class as target
  (create-DbPatch (:track_id cRec) (:datetime cRec) (:path cRec) (:action cRec) (:patchkey cRec) (:value cRec)))

(defn DbPatch-to-java [cRec]
	(DbPatch. (:id cRec) (:track_id cRec) (:datetime cRec) (:path cRec) (:action cRec) (:patchkey cRec) (:value cRec) ))


(add-to-java (DbPatch_clj. 0 1 2 3 4 5 6) DbPatch-to-java)

(defn DbPatch-to-clj [jRec]
	(DbPatch_clj. (.getId jRec) (.getTrack_id jRec) (.getDatetime jRec) (.getPath jRec) (.getAction jRec) (.getPatchkey jRec) (.getValue jRec) ))


(add-to-clj (DbPatch.) DbPatch-to-clj)

